### PR TITLE
Implement websocket reconnect feature and display status on all pages

### LINKED ISSUE #62: Handle connection loss
At the moment when the connection is lost, a user will not notice it and can even perform updates that will not be persisted. This is not good. I propose two changes:
* Give a visual cue that the connection is lost.
  * At the moment on the login screen, the user will see "Connecting..." which is likely good enough here
    blocking
  * On the voting page, we could similarly replace the "Reveal" button with a grey "Connecting..." button
  * On the results page, we could do the same with the "Reset" button
* Automatically try to reconnect
  * Something like this looks reasonable https://stackoverflow.com/a/23176223/11576601
  * Additionally, we might use an exponential backoff https://en.wikipedia.org/wiki/Exponential_backoff strategy to reduce the reconnection rate on failure

An important point is to check how the app behaves when more than one window is open. I noticed that opening a second window but logging in as the same user will kill the first connection. In combination with automatic reconnection, we have a problem. Either we prevent logging in with the same name (not so nice) or find a way to have more than one browser controlling the same name simultaneously (very nice!)

### PR DESCRIPTION
**Pull request by ThiloAschebrock**
This PR closes #62.

Changes:

- Display "Connecting..." on reveal and reset button if the websocket connection is present.
- Implement automatic reconnect feature for the websocket with exponential backoff.

image
image

Open point:

- [x] Behavior when more then one window is open:

> An important point is to check how the app behaves when more than one window is open. I noticed that opening a second window but logging in as the same user will kill the first connection. In combination with automatic reconnection, we have a problem. Either we prevent logging in with the same name (not so nice) or find a way to have more than one browser controlling the same name simultaneously (very nice!)

### CODE REVIEW

**Comment by lukastaegert** [comment]
Awesome you dig into this! Will try to do a proper review in the next days, meanwhile some thoughts about the open question, that I also had on my mind for some time.

So what is the problem we are solving:
- When the connection is terminated for technical reasons (bad internet), we want to reconnect as soon as possible

However the issue is, that when the connection is terminated because another user/browser tab enters the same name, we probably want one of the following:
1. Have two windows share a session, or
2. Drop the initial session to the login screen, possibly with a message indicating another session was started with the same name

The first has some technical obstacles, but is also a bad choice because it would allow people to "spy" on other peoples sessions without them noticing.

So the easiest choice could be that when a new login is performed for an existing session, the existing window gets a special "logged-out" message that it can react to.

But seeing as at the moment, a new login will just silently steal the session, the implemented behaviour in this PR is already a net improvement.

**Comment by ThiloAschebrock** [comment]
> So the easiest choice could be that when a new login is performed for an existing session, the existing window gets a special "logged-out" message that it can react to.

I implemented a special logout message in #94 if a user is kicked, I was able to reused it here with a few lines of code:
image

**Comment by lukastaegert** [comment]
That sounds perfect!

**Review by lukastaegert — APPROVED** [review]
Looks great, just one stylistic comment

File: frontend/src/Components/RevealButton/RevealButton.tsx [hunk 1]
```diff
@@ -12,21 +12,32 @@ export const RevealButton = connectToWebSocket(
...
-    }
     return (
-      <button class={classes.revealButton} onClick={revealVotes}>
-        {BUTTON_REVEAL_VOTES}
+      <button class={classes.revealButton} onClick={revealVotes} disabled={!connected}>
+        {renderContent()}
       </button>
     );
+
+    function renderContent() {
```
[the remark this answers is not in the document]
Comment (ThiloAschebrock) [reply frontend/src/Components/RevealButton/RevealButton.tsx]: Done as suggested in 4b93473.

**Review by lukastaegert — APPROVED** [review]

**Pull request merged**